File: LibTerm/Commands/python3.m
//
//  Python.c
//  LibTerm
//
//  Created by Adrian Labbe on 12/14/18.
//  Copyright © 2018 Adrian Labbe. All rights reserved.
//

#include "../../Python/Headers/Python.h"
#include "Py_UnixMain.h"
#import <Foundation/Foundation.h>
#import <ios_system/ios_system.h>

int python3_main(int argc, char **argv) {
    
    const char * pyPath = [[NSString stringWithUTF8String:"PYTHONPATH="] stringByAppendingString:[NSBundle.mainBundle pathForResource:@"python37" ofType:@"zip"]].UTF8String;
    
    FILE* oldStdin = stdin;
    FILE* oldStdout = stdout;
    FILE* oldStderr = stderr;
    
    stdin = thread_stdin;
    stdout = thread_stdout;
    stderr = stdout;
    
    if (pyPath) {
        putenv(pyPath);
    }
    
    int py = _Py_UnixMain(argc, argv);
    
    stdin = oldStdin;
    stdout = oldStdout;
    stderr = oldStderr;
    
    return py;
}

File: LibTerm/Commands/python3.swift
//
//  python3.swift
//  LibTerm
//
//  Created by Adrian Labbé on 05.01.19.
//  Copyright © 2019 Adrian Labbe. All rights reserved.
//

import Foundation

/// The `python` command.
@_cdecl("python3_swift_main") public func python3_swift_main(argc: Int32, argv: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>!) -> Int32 {
    return _Py_UnixMain(argc, argv)
}
